test/automation/src/terminal.ts

```diff
@@ -51,42 +75,64 @@ export class Terminal {
 		await this.code.dispatchKeybinding('enter');
 	}
 
-	// TODO: Return something more robust:
-	// export interface ITerminalInstance {
-	//   name: string;
-	//   icon: string;
-	// }
-	// export type TerminalGroup = ITerminalInstance[];
-	// export type TerminalLayout = TerminalGroup[];
-	async getTabLabels(expectedCount: number, splits?: boolean, accept?: (result: IElement[]) => boolean): Promise<string[]> {
-		const result: string[] = [];
-		const tabs = await this.code.waitForElements(TABS, true, e => accept ? accept(e) : e.length === expectedCount && (!splits || e.some(e => e.textContent.startsWith('┌'))) && e.every(element => element.textContent.trim().length > 1));
-		for (const t of tabs) {
-			result.push(t.textContent);
-		}
-		if (!result[0].startsWith('┌')) {
-			const first = result[1];
-			const second = result[0];
-			return [first, second];
+	async assertEditorGroupCount(count: number): Promise<void> {
+		await this.code.waitForElements(EDITOR_GROUPS_SELECTOR, true, editorGroups => editorGroups && editorGroups.length === count);
+	}
+
+	async assertSingleTab(label: TerminalLabel, editor?: boolean): Promise<void> {
+		const selector = editor ? EDITOR_TAB_SELECTOR : SINGLE_TAB_SELECTOR;
+		const element = await this.code.waitForElement(selector, elt => elt ? elt.textContent.trim().length > 1 : false);
```

Comment:
to do move the call to `waitForElement `into `assertTabExpected `so that the icon, name, and color are all awaited for in one big conditional

like `await this.code.waitForElement(selector, elt => elt ? elt.textContent.match(regexp considering name and split decoration)  && elt.children.find(c => c.className.includes(color) && elt.className.includes(icon));`